src/tests/test_magic.py

```diff
@@ -195,6 +195,162 @@ def test_persist_bare(ip):
+            "website",
+            [
+                (
+                    0,
+                    "Bertold Brecht",
+                    "https://en.wikipedia.org/wiki/Bertolt_Brecht",
+                    1954,
+                )
+            ],
+        ),
+        ("number_table", [(0, 4, -2)]),
+    ],
+)
+def test_persist_replace_no_override(ip, test_table, expected_result):
+    # New saved dataframe length -> 1
+    saved_df_name = query_and_save_as_dataframe(ip, table_name=test_table, limit=1)
+    ip.run_cell(f"%sql --persist-replace sqlite:// {saved_df_name}")
+    out = ip.run_cell(f"%sql SELECT * FROM {saved_df_name}")
+    assert out.result == expected_result
+    assert out.error_in_exec is None
+
+
+@pytest.mark.parametrize(
+    "test_table, expected_result",
+    [
+        ("test", [(0, 1, "foo")]),
+        ("author", [(0, "William", "Shakespeare", 1616)]),
+        (
+            "website",
+            [
+                (
+                    0,
+                    "Bertold Brecht",
+                    "https://en.wikipedia.org/wiki/Bertolt_Brecht",
+                    1954,
+                )
+            ],
+        ),
+        ("number_table", [(0, 4, -2)]),
+    ],
+)
+def test_persist_replace_override(ip, test_table, expected_result):
+    # Previous saved dataframe length -> 2
+    saved_df_name = query_and_save_as_dataframe(ip, table_name=test_table, limit=2)
+    ip.run_cell(f"%sql --persist sqlite:// {saved_df_name}")
+    # New saved dataframe length -> 1
+    saved_df_name = query_and_save_as_dataframe(ip, table_name=test_table, limit=1)
+    ip.run_cell(f"%sql --persist-replace sqlite:// {saved_df_name}")
+
+    out = ip.run_cell(f"%sql SELECT * FROM {saved_df_name}")
+    # To test it's referring to new saved dataframe
+    assert out.result == expected_result
+    assert out.error_in_exec is None
+
+
+@pytest.mark.parametrize(
+    "test_table", [("test"), ("author"), ("website"), ("number_table")]
+)
+def test_persist_replace_override_reverted_order(ip, test_table):
+    # Previous saved dataframe length -> 2
+    saved_df_name = query_and_save_as_dataframe(ip, table_name=test_table, limit=2)
+    ip.run_cell(f"%sql --persist-replace sqlite:// {saved_df_name}")
+
+    # New saved dataframe length -> 1
+    saved_df_name = query_and_save_as_dataframe(ip, table_name=test_table, limit=1)
+    out = ip.run_cell(f"%sql --persist sqlite:// {saved_df_name}")
+
+    assert "Table already exist, maybe use --persist-replace" in str(out.error_in_exec)
+
+
+@pytest.mark.parametrize(
+    "test_table", [("test"), ("author"), ("website"), ("number_table")]
+)
+def test_persist_and_append_use_together(ip, test_table):
+    # Test error message when use --persist and --persist-replace together
+    saved_df_name = query_and_save_as_dataframe(ip, table_name=test_table, limit=1)
+    out = ip.run_cell(f"%sql --persist-replace --append sqlite:// {saved_df_name}")
+
+    # assert isinstance(out.error_in_exec, ValueError)
```

Comment:
Let's remove this line and check the name of the error as well (which should be "ValueError")